server/py/services/api/initial_data.py

```diff
@@ -993,6 +999,57 @@ def _create_project_summaries(db, db_session):
     db._upsert(db_session, project_summaries, ignore=True)
 
 
+def _init_system_id(db_session: sqlalchemy.orm.Session):
+    """
+    Initializes a system id for MLRun deployment.
+    The system id is first checked in the database. If it does not exist, the function checks if an id was set in the
+    config, and if neither is found, a new random one is generated and stored.
+    """
+
+    db = framework.db.sqldb.db.SQLDB()
+
+    # check if a system id already exists in the database
+    system_id = db.get_system_id(db_session)
+
+    if system_id is not None:
+        logger.debug("Existing system id found in the database", system_id=system_id)
+        mlrun.mlconf.system_id = system_id
+        return
+
+    logger.debug("System id not found in DB")
+    # check if the system id is already set in the config
+    system_id = _get_configured_system_id()
+
+    if system_id:
+        logger.debug("Using configured system id", system_id=system_id)
+    else:
+        # if no system id is found, generate a new one
+        system_id = _generate_system_id()
+    db.store_system_id(db_session, system_id)
+
+    # set the system id in mlrun config
+    mlrun.mlconf.system_id = system_id
+
+    logger.info("Initialized system ID", system_id=system_id)
+
+
+def _get_configured_system_id() -> typing.Optional[str]:
+    return mlrun.mlconf.system_id or None
+
+
+def _generate_system_id() -> str:
+    # Generate a random 32-bit unsigned integer and encode as a URL-safe Base64 string without padding
+    while True:
+        random_number = random.getrandbits(32)
+        random_bytes = random_number.to_bytes(4, "big")
+        base64_str = base64.urlsafe_b64encode(random_bytes).decode("utf-8").rstrip("=")
+
+        # ensure the string does not start with '-' or '_', as these characters are not allowed at the start of DNS
+        # names or Kubernetes resource names
+        if not base64_str.startswith(("-", "_")):
+            return base64_str
```

Comment:
@Yacouby this implementation is case sensitive, but for names it is better to have all lowercase so we won't have 2 ids that are different in case.

Also, underscore (`_`) is not allowed in kubernetes qualified name. You can use the regex Liran suggested - https://github.com/mlrun/mlrun/blob/196234944cd6d95eeaa81e9654a099ff8a4638d1/mlrun/utils/regex.py#L26